pkg/oidcclient/login.go

```diff
@@ -140,7 +140,7 @@ func WithLogger(logger logr.Logger) Option {
 // system at the time of the request.
 func WithListenPort(port uint16) Option {
 	return func(h *handlerState) error {
-		h.listenAddr = fmt.Sprintf("localhost:%d", port)
+		h.listenAddr = net.JoinHostPort("localhost", fmt.Sprint(port))
```

Comment:
Same comment about tests.

Reply:
The existing tests cover this. The change was more forward thinking, i.e if `localhost` were ever changed to allow IPv6 addresses. We can change this back if you'd prefer.

Reply:
In this case I'm not sure this change is needed (since `"localhost"` is just a string constant and can't be an IPv6 address), but it's more correct and it doesn't hurt.